src/de/grobox/liberario/activities/MainActivity.java

```diff
@@ -114,7 +114,11 @@ public void onCreate(Bundle savedInstanceState) {
              .withOnAccountHeaderListener(new AccountHeader.OnAccountHeaderListener() {
 	             @Override
 	             public boolean onProfileChanged(View view, IProfile profile, boolean currentProfile) {
-		             if(!currentProfile && profile != null && profile instanceof ProfileDrawerItem) {
+		             if(currentProfile) {
+			             Intent intent = new Intent(getContext(), PickNetworkProviderActivity.class);
+			             ActivityOptionsCompat options = ActivityOptionsCompat.makeScaleUpAnimation(getCurrentFocus(), 0, 0, 0, 0);
+			             ActivityCompat.startActivityForResult(MainActivity.this, intent, CHANGED_NETWORK_PROVIDER, options.toBundle());
+		             } else if(profile != null && profile instanceof ProfileDrawerItem) {
```

Comment:
Thanks for the review. You're right. I added a `openPickNetworkProviderActivity()` method. I hope this is ok.

I also return `true` now. I don't see any change but I don't know what it means to consume the event (I'm an android newby 🙈 )